src/Numerics/Optimization/NewtonMinimizer.cs

```diff
@@ -125,6 +128,46 @@ public static MinimizationResult Minimum(IObjectiveFunction objective, Vector<do
             return new MinimizationWithLineSearchResult(objective, iterations, ExitCondition.AbsoluteGradient, totalLineSearchSteps, iterationsWithNontrivialLineSearch);
         }
 
+        static Vector<double> CalculateSearchDirection(IObjectiveFunction objective, HessianModifiers modifier)
+        {
+            Vector<double> searchDirection = null;
+            switch (modifier)
+            {
+                case HessianModifiers.None:
+                    searchDirection = SolveLU(objective);
+                    break;
+                case HessianModifiers.ReverseNegativeEigenValues:
+                    searchDirection = ReverseNegativeEigenValuesAndSolve(objective);
+                    break;
+            }
+
+            return searchDirection;
+        }
+
+        static Vector<double> SolveLU(IObjectiveFunction objective)
+        {
+            return objective.Hessian.LU().Solve(-objective.Gradient);
+        }
+
+        /// <summary>
+        /// Force the Hessian to be positive definite by reversing the negative eigenvalues. Use the EVD decomposition to then solve for the search direction.
+        /// Implementation based on Philip E. Gill, Walter Murray, and Margaret H. Wright, Practical Optimization, 1981, 107–8.
+        /// </summary>
+        /// <param name="objective"></param>
+        /// <returns></returns>
+        static Vector<double> ReverseNegativeEigenValuesAndSolve(IObjectiveFunction objective)
+        {
+            Evd<double> evd = objective.Hessian.Evd(Symmetricity.Symmetric);
+            for (int i = 0; i < evd.EigenValues.Count; i++)
+            {
+                if (evd.EigenValues[i].Real < double.Epsilon)
+                {
+                    evd.EigenValues[i] = Math.Max(-evd.EigenValues[i].Real, double.Epsilon);
```

Comment:
You can skip the `if` condition altogether, if you write this as `evd.EigenValues[i] = Math.Max(Math.Abs(evd.EigenValues[i].Real), double.Epsilon);`

Reply:
@jkalias I am sorry I didn't see your very helpful review comments until much later. I accepted your proposal in this comment and all others in the latest commit to this PR. Thank you for taking the time to review!